utils/doubly_linked_list.ml

```diff
@@ -340,6 +340,14 @@ let for_all t ~f =
   in
   aux t f t.first
 
+let for_all_i t ~f =
+  let rec aux t f i curr =
```

Comment:
`t` does not seem to be used by `aux`
(except for the recursive call).

Reply:
Removed here and in `for_all`.